docs/CODING_STYLE.md

```diff
@@ -84,7 +86,25 @@ layout: default
 
 - Do not write functions that clobber call-by-reference variables on
   failure. Use temporary variables for these cases and change the passed in
-  variables only on success.
+  variables only on success. The rule is: never clobber return parameters on
+  failure, always initialize return parameters on success.
+
+- Typically, function parameters fit into three categories: input parameters,
+  mutable objects, and call-by-reference return parameters. Input parameters
+  should always carry suitable "const" declarators if they are pointers, to
+  indicate they are input-only and not changed by the functional. Return
+  parameters are best prefixed with "ret_", to clarify they are return
+  parameters. Example:
+
+  ```c
+  static int foobar_frobnicate(
+                  Foobar* object,            /* the associated mutable object */
+                  const char *input,         /* immutable input parameter */
+                  char **ret_frobnicated) {  /* return parameter */
+          …
+          return 0;
+  }
```

Comment:
Hmm, maybe say that parameters which are not mutable objects, but are input-output parameters should not be prefixed with `ret_` (e.g. when there's a counter or `should_indent` boolean or stuff like that).